src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -436,6 +438,16 @@ module.exports = function<T, P, I, TI, C, CX>(
     } else {
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
+      if (__DEV__) {
+        warning(
+          !!workInProgress.ref,
+          'Stateless function components cannot be given refs ' +
+          '(See ref "' + workInProgress.ref._stringRef + '" in ' +
```

Comment:
There is no test case because we silently give you `null`. It's bad though.

I'd prefer we keep warning parity for now. So if we add it to Fiber we should also backport it to Stack. See https://github.com/facebook/react/pull/8635#issuecomment-269092380 for context.